pkg/util/instancemetadata/prod.go

```diff
@@ -71,7 +72,7 @@ func (p *prod) populateTenantIDFromMSI(ctx context.Context) error {
 	c := &azureclaim.AzureClaim{}
 	_, _, err = parser.ParseUnverified(token.OAuthToken(), c)
 	if err != nil {
-		return err
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidServicePrincipalToken, "properties.servicePrincipalProfile", "The provided service principal generated an invalid token.")
```

Comment:
It doesn't make sense to switch this error into an CloudError because the code path here is during RP instancemeta initialisation and there is no API request involved.

Reply:
Reverted the changes, thank you.